File: 프로그래머스/0/181924. 수열과 구간 쿼리 3/수열과 구간 쿼리 3.cpp
#include <string>
#include <vector>

using namespace std;

void swap2(int & a, int & b) {
    int tmp = 0;
    tmp = a;
    a = b;
    b = tmp;
}

vector<int> solution(vector<int> arr, vector<vector<int>> queries) {
    vector<int> answer;
    for (int q = 0; q < queries.size(); q++) {
        int i = queries[q][0]; // 쿼리의 첫 번째 원소 (인덱스 i)
        int j = queries[q][1]; // 쿼리의 두 번째 원소 (인덱스 j)
        
        // 2. arr[i]와 arr[j]의 값을 서로 바꿈
        swap2(arr[i], arr[j]);
    }
    answer = arr;
    return answer;
}

File: 프로그래머스/0/181924. 수열과 구간 쿼리 3/수열과 구간 쿼리 3.py
def solution(arr, queries):
    answer = []
    
    for idx in range(len(queries)):
        a = queries[idx][0]
        b = queries[idx][1]
        
        tmp = arr[a]
        arr[a] = arr[b]
        arr[b] = tmp
        
    answer = arr;
    return answer